module/mempool/queue/chunkDataPack.go

```diff
@@ -0,0 +1,120 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/mempool"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+)
+
+// ChunkDataPackRequestQueue implements a HeroCache-based in-memory queue for storing chunk data pack requests.
+// It is designed to be utilized at Execution Nodes to maintain and respond chunk data pack requests.
+type ChunkDataPackRequestQueue struct {
+	mu        sync.RWMutex
+	cache     *herocache.Cache
+	sizeLimit uint
+}
+
+var _ mempool.ChunkDataPackRequestQueue = &ChunkDataPackRequestQueue{}
+
+func NewChunkDataPackRequestQueue(sizeLimit uint32, logger zerolog.Logger, collector module.HeroCacheMetrics) *ChunkDataPackRequestQueue {
+	return &ChunkDataPackRequestQueue{
+		cache: herocache.NewCache(
+			sizeLimit,
+			herocache.DefaultOversizeFactor,
+			heropool.LRUEjection,
```

Comment:
given that this is a queue, should it be LIFO?  Or maybe, given that size is fixed, we can have a no-ejection mode?

Reply:
https://github.com/onflow/flow-go/pull/2951/commits/887d541421ea5a7e7e5b493dbd2318058d72d61a
https://github.com/onflow/flow-go/pull/2951/commits/4bbeb862a6b17538d04c6091f138d27dbd11deac
https://github.com/onflow/flow-go/pull/2951/commits/bcb19eb60a7e9fd59f5712883ac35fe57c9a462c